### PR TITLE
use DataTree from xarray instead of from xarray-datatree

### PR DESCRIPTION
**Pull request by OriolAbril**
related to #14. I wouldn't merge just yet, and instead merge as many PRs as possible,
create similar PRs in arviz-stats and arviz-plots to pull the band aid.

----
📚 Documentation preview 📚: https://arviz-base--24.org.readthedocs.build/en/24/

### CODE REVIEW

**Review by amaloney — APPROVED** [review]
LGTM

I would not merge this to `main` though, since it has breaking changes. Let's make a "0.3.0" branch and merge to that. Then we can sync with the other packages using the 0.3.0 branch to fix problems. For each project, we should make a "0.3.0" branch as well, and any fixes that go into them using this branch, should be merged into their respective 0.3.0 branches. Once all bugs have been ironed out, we can push each 0.3.0 branch into main, tag, and then release.

**Comment by OriolAbril** [comment]
I was thinking about making a 0.3 release before merging, then using 0.4.0.dev0 as version on main and merging this. But at this stage I would not worry too much about releases nor breaking changes. For https://github.com/arviz-devs/arviz-stats/issues/16 for example my idea was also to directly do the change in arviz-stats, update arviz-plots as needed and consider it done, no warnings or duplication of dim/dims

**Pull request merged**
